primary/src/primary.rs

```diff
@@ -207,10 +208,28 @@ impl Primary {
         let routes = anemo::Router::new()
             .add_rpc_service(primary_service)
             .add_rpc_service(worker_service);
+
+        let service = ServiceBuilder::new()
+            .layer(TraceLayer::new())
```

Comment:
Is it possible/do you think it's worthwhile to add a `MystenNetworkLayer` type of thing that itself includes all of: `TraceLayer`, the metrics `CallbackLayer`, and ownership of the actual metrics struct? This means future changes/additions to our standard suite of layers can go just in that one place instead of many places, plus reduces some boilerplate in these main functions.

Reply:
Yes we could probably look at creating a simple function that returns the layer stack we need, see here for an example of how we could do that: https://docs.rs/tower/latest/tower/util/struct.BoxLayer.html

Reply:
But i'm going to leave this to be done at a future point in time.